// @zoralabs/v3/dist/typechain/factories/ZoraProtocolFeeSettings.sol/IERC721TokenURI__factory.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */

import { Contract, Signer, utils } from "ethers";
import type { Provider } from "@ethersproject/providers";
import type {
  IERC721TokenURI,
  IERC721TokenURIInterface,
} from "../../ZoraProtocolFeeSettings.sol/IERC721TokenURI";

const _abi = [
  {
    inputs: [
      {
        internalType: "uint256",
        name: "tokenId",
        type: "uint256",
      },
    ],
    name: "tokenURI",
    outputs: [
      {
        internalType: "string",
        name: "",
        type: "string",
      },
    ],
    stateMutability: "view",
    type: "function",
  },
];

export class IERC721TokenURI__factory {
  static readonly abi = _abi;
  static createInterface(): IERC721TokenURIInterface {
    return new utils.Interface(_abi) as IERC721TokenURIInterface;
  }
  static connect(
    address: string,
    signerOrProvider: Signer | Provider
  ): IERC721TokenURI {
    return new Contract(address, _abi, signerOrProvider) as IERC721TokenURI;
  }
}
