// @zoralabs/nft-hooks/dist/types/NFTInterface.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FIXED_PRICE_EVENT_TYPES = exports.AUCTION_EVENT_TYPES = exports.FIXED_SIDE_TYPES = exports.AUCTION_SOURCE_TYPES = exports.TOKEN_TRANSFER_EVENT_CONTEXT_TYPES = exports.FIXED_PRICE_MARKET_SOURCES = exports.TOKEN_TRANSFER_EVENT_TYPES = exports.MEDIA_SOURCES = exports.MARKET_INFO_STATUSES = exports.EDITION_SOURCES = exports.MARKET_TYPES = exports.KNOWN_CONTRACTS = void 0;
var KNOWN_CONTRACTS;
(function (KNOWN_CONTRACTS) {
    KNOWN_CONTRACTS["ZORA"] = "zora";
})(KNOWN_CONTRACTS = exports.KNOWN_CONTRACTS || (exports.KNOWN_CONTRACTS = {}));
var MARKET_TYPES;
(function (MARKET_TYPES) {
    MARKET_TYPES["AUCTION"] = "Auction";
    MARKET_TYPES["FIXED_PRICE"] = "FixedPrice";
    MARKET_TYPES["EDITION"] = "Edition";
})(MARKET_TYPES = exports.MARKET_TYPES || (exports.MARKET_TYPES = {}));
var EDITION_SOURCES;
(function (EDITION_SOURCES) {
    EDITION_SOURCES["CUSTOM"] = "Custom";
    EDITION_SOURCES["ZORA_EDITIONS"] = "ZoraEditions";
})(EDITION_SOURCES = exports.EDITION_SOURCES || (exports.EDITION_SOURCES = {}));
var MARKET_INFO_STATUSES;
(function (MARKET_INFO_STATUSES) {
    MARKET_INFO_STATUSES["PENDING"] = "pending";
    MARKET_INFO_STATUSES["ACTIVE"] = "active";
    MARKET_INFO_STATUSES["COMPLETE"] = "complete";
    MARKET_INFO_STATUSES["CANCELED"] = "canceled";
    MARKET_INFO_STATUSES["UNKNOWN"] = "unknown";
    MARKET_INFO_STATUSES["INVALID"] = "invalid";
})(MARKET_INFO_STATUSES = exports.MARKET_INFO_STATUSES || (exports.MARKET_INFO_STATUSES = {}));
var MEDIA_SOURCES;
(function (MEDIA_SOURCES) {
    MEDIA_SOURCES["OPENSEA"] = "opensea";
    MEDIA_SOURCES["ZORA"] = "zora";
    MEDIA_SOURCES["DERIVED"] = "derived";
    MEDIA_SOURCES["RAW"] = "raw";
})(MEDIA_SOURCES = exports.MEDIA_SOURCES || (exports.MEDIA_SOURCES = {}));
var TOKEN_TRANSFER_EVENT_TYPES;
(function (TOKEN_TRANSFER_EVENT_TYPES) {
    TOKEN_TRANSFER_EVENT_TYPES["MINT"] = "mint";
    TOKEN_TRANSFER_EVENT_TYPES["BURN"] = "burn";
    TOKEN_TRANSFER_EVENT_TYPES["TRANSFER"] = "transfer";
    TOKEN_TRANSFER_EVENT_TYPES["SALE"] = "sale";
})(TOKEN_TRANSFER_EVENT_TYPES = exports.TOKEN_TRANSFER_EVENT_TYPES || (exports.TOKEN_TRANSFER_EVENT_TYPES = {}));
var FIXED_PRICE_MARKET_SOURCES;
(function (FIXED_PRICE_MARKET_SOURCES) {
    FIXED_PRICE_MARKET_SOURCES["ZNFT_PERPETUAL"] = "ZNFTPerpetual";
    FIXED_PRICE_MARKET_SOURCES["ZORA_ASK_V1"] = "ZoraAskV1";
    FIXED_PRICE_MARKET_SOURCES["ZORA_ASK_V1_EVENT"] = "ZoraAskV1Event";
    FIXED_PRICE_MARKET_SOURCES["ZORA_ASK_V3"] = "ZoraAskV3";
    FIXED_PRICE_MARKET_SOURCES["OPENSEA_FIXED"] = "OpenseaFixed";
    FIXED_PRICE_MARKET_SOURCES["FOUNDATION_FIXED"] = "FoundationFixed";
})(FIXED_PRICE_MARKET_SOURCES = exports.FIXED_PRICE_MARKET_SOURCES || (exports.FIXED_PRICE_MARKET_SOURCES = {}));
var TOKEN_TRANSFER_EVENT_CONTEXT_TYPES;
(function (TOKEN_TRANSFER_EVENT_CONTEXT_TYPES) {
    TOKEN_TRANSFER_EVENT_CONTEXT_TYPES["TOKEN_TRANSFER_EVENT"] = "TokenTransferEvent";
    TOKEN_TRANSFER_EVENT_CONTEXT_TYPES["TOKEN_MARKET_EVENT"] = "TokenMarketEvent";
})(TOKEN_TRANSFER_EVENT_CONTEXT_TYPES = exports.TOKEN_TRANSFER_EVENT_CONTEXT_TYPES || (exports.TOKEN_TRANSFER_EVENT_CONTEXT_TYPES = {}));
var AUCTION_SOURCE_TYPES;
(function (AUCTION_SOURCE_TYPES) {
    AUCTION_SOURCE_TYPES["ZORA_RESERVE_V2"] = "ZoraReserveV2";
    AUCTION_SOURCE_TYPES["OPENSEA_ENGLISH"] = "OpenseaEnglish";
})(AUCTION_SOURCE_TYPES = exports.AUCTION_SOURCE_TYPES || (exports.AUCTION_SOURCE_TYPES = {}));
var FIXED_SIDE_TYPES;
(function (FIXED_SIDE_TYPES) {
    FIXED_SIDE_TYPES["ASK"] = "ask";
    FIXED_SIDE_TYPES["OFFER"] = "offer";
})(FIXED_SIDE_TYPES = exports.FIXED_SIDE_TYPES || (exports.FIXED_SIDE_TYPES = {}));
var AUCTION_EVENT_TYPES;
(function (AUCTION_EVENT_TYPES) {
    AUCTION_EVENT_TYPES["AUCTION_CREATED"] = "AuctionCreated";
    AUCTION_EVENT_TYPES["AUCTION_BID"] = "AuctionBid";
    AUCTION_EVENT_TYPES["AUCTION_ENDED"] = "AuctionEnded";
    AUCTION_EVENT_TYPES["AUCTION_FINALIZED"] = "AuctionFinalized";
    AUCTION_EVENT_TYPES["AUCTION_APPROVED"] = "AuctionApproved";
    AUCTION_EVENT_TYPES["AUCTION_CANCELLED"] = "AuctionCancelled";
    AUCTION_EVENT_TYPES["AUCTION_UPDATED"] = "AuctionUpdated";
})(AUCTION_EVENT_TYPES = exports.AUCTION_EVENT_TYPES || (exports.AUCTION_EVENT_TYPES = {}));
var FIXED_PRICE_EVENT_TYPES;
(function (FIXED_PRICE_EVENT_TYPES) {
    FIXED_PRICE_EVENT_TYPES["FIXED_PRICE_CREATED"] = "FixedPriceCreated";
    FIXED_PRICE_EVENT_TYPES["FIXED_PRICE_FILLED"] = "FixedPriceFilled";
    FIXED_PRICE_EVENT_TYPES["FIXED_PRICE_CANCELLED"] = "FixedPriceCancelled";
    FIXED_PRICE_EVENT_TYPES["FIXED_PRICE_UPDATED"] = "FixedPriceUpdated";
})(FIXED_PRICE_EVENT_TYPES = exports.FIXED_PRICE_EVENT_TYPES || (exports.FIXED_PRICE_EVENT_TYPES = {}));
