// @zoralabs/nft-hooks/dist/types/NFTQuery.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ViewType = exports.SortField = exports.SortDirection = exports.MarketType = void 0;
var MarketType;
(function (MarketType) {
    MarketType[MarketType["FIXED_PRICE"] = 0] = "FIXED_PRICE";
    MarketType[MarketType["ANY_MARKET"] = 1] = "ANY_MARKET";
    MarketType[MarketType["AUCTION"] = 2] = "AUCTION";
    MarketType[MarketType["SOLD"] = 3] = "SOLD";
})(MarketType = exports.MarketType || (exports.MarketType = {}));
var SortDirection;
(function (SortDirection) {
    SortDirection[SortDirection["ASC"] = 0] = "ASC";
    SortDirection[SortDirection["DESC"] = 1] = "DESC";
})(SortDirection = exports.SortDirection || (exports.SortDirection = {}));
var SortField;
(function (SortField) {
    SortField[SortField["ACTIVE"] = 0] = "ACTIVE";
    SortField[SortField["AUCTION_PRICE"] = 1] = "AUCTION_PRICE";
    SortField[SortField["FIXED_PRICE"] = 2] = "FIXED_PRICE";
    SortField[SortField["ANY_PRICE"] = 3] = "ANY_PRICE";
    SortField[SortField["TOKEN_ID"] = 4] = "TOKEN_ID";
    SortField[SortField["MINTED"] = 5] = "MINTED";
})(SortField = exports.SortField || (exports.SortField = {}));
var ViewType;
(function (ViewType) {
    ViewType[ViewType["SUMMARY"] = 0] = "SUMMARY";
    ViewType[ViewType["FULL"] = 1] = "FULL";
})(ViewType = exports.ViewType || (exports.ViewType = {}));
