// @zoralabs/nft-hooks/dist/backends/zora-user/ZoraUserDataSource.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ZoraUserDataSource = void 0;
const tslib_1 = require("tslib");
const dataloader_1 = tslib_1.__importDefault(require("dataloader"));
const urls_1 = require("../../constants/urls");
const FetchWithTimeout_1 = require("../../fetcher/FetchWithTimeout");
class ZoraUserDataSource {
    constructor(timeout = 4) {
        this.loadProfile = async (address) => {
            return await this.userLoader.load(address.toLowerCase());
        };
        this.loadProfiles = async (addresses) => {
            return await this.userLoader.loadMany(addresses.map((address) => address.toLowerCase()));
        };
        this.fetchZoraUsers = async (addresses) => {
            const fetchWithTimeout = new FetchWithTimeout_1.FetchWithTimeout(this.timeout);
            const response = await fetchWithTimeout.fetch(urls_1.ZORA_USERNAME_API_URL, {
                method: 'POST',
                type: 'cors',
                headers: {
                    'content-type': 'application/json',
                },
            });
            const usernames = (await response.json());
            return addresses.map((address) => {
                return usernames.find((user) => user.address === address) || { address };
            });
        };
        this.userLoader = new dataloader_1.default(this.fetchZoraUsers);
        this.timeout = timeout;
    }
}
exports.ZoraUserDataSource = ZoraUserDataSource;
