// @zoralabs/v3/dist/typechain/ReserveAuctionFindersEth.t.sol/ReserveAuctionFindersEthTest.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
import type {
  BaseContract,
  BigNumber,
  BytesLike,
  CallOverrides,
  ContractTransaction,
  Overrides,
  PopulatedTransaction,
  Signer,
  utils,
} from "ethers";
import type {
  FunctionFragment,
  Result,
  EventFragment,
} from "@ethersproject/abi";
import type { Listener, Provider } from "@ethersproject/providers";
import type {
  TypedEventFilter,
  TypedEvent,
  TypedListener,
  OnEvent,
  PromiseOrValue,
} from "../common";

export interface ReserveAuctionFindersEthTestInterface extends utils.Interface {
  functions: {
    "IS_TEST()": FunctionFragment;
    "failed()": FunctionFragment;
    "setUp()": FunctionFragment;
    "testRevert_AuctionNotOver()": FunctionFragment;
    "testRevert_AuctionNotStarted()": FunctionFragment;
    "testRevert_BidMustBe10PercentGreaterThanPrevious()": FunctionFragment;
    "testRevert_BidMustMeetReservePrice()": FunctionFragment;
    "testRevert_CannotBidOnAuctionNotActive()": FunctionFragment;
    "testRevert_CannotBidOnAuctionNotStarted()": FunctionFragment;
    "testRevert_CannotBidOnExpiredAuction()": FunctionFragment;
    "testRevert_CannotCancelActiveAuction()": FunctionFragment;
    "testRevert_CannotUpdateActiveAuction()": FunctionFragment;
    "testRevert_CannotUpdateAuctionDoesNotExist()": FunctionFragment;
    "testRevert_FindersFeeBPSCannotExceed10000()": FunctionFragment;
    "testRevert_InvalidTransferBeforeFirstBid()": FunctionFragment;
    "testRevert_MustApproveModule()": FunctionFragment;
    "testRevert_MustBeTokenOwnerOrOperator()": FunctionFragment;
    "testRevert_MustSpecifySellerFundsRecipient()": FunctionFragment;
    "testRevert_OnlySellerOrOwnerCanCancel()": FunctionFragment;
    "testRevert_SellerMustApproveERC721TransferHelper()": FunctionFragment;
    "testRevert_UpdateMustBeSeller()": FunctionFragment;
    "test_CancelAuction()": FunctionFragment;
    "test_CreateAuction()": FunctionFragment;
    "test_CreateAuctionAndCancelPrevious()": FunctionFragment;
    "test_CreateFirstBid()": FunctionFragment;
    "test_CreateFutureAuction()": FunctionFragment;
    "test_ExtendAuction()": FunctionFragment;
    "test_RefundPreviousBidder()": FunctionFragment;
    "test_SetReservePrice()": FunctionFragment;
    "test_SettleAuction()": FunctionFragment;
    "test_StoreTimeOfFirstBid()": FunctionFragment;
    "test_TransferNFTIntoEscrow()": FunctionFragment;
  };

  getFunction(
    nameOrSignatureOrTopic:
      | "IS_TEST"
      | "failed"
      | "setUp"
      | "testRevert_AuctionNotOver"
      | "testRevert_AuctionNotStarted"
      | "testRevert_BidMustBe10PercentGreaterThanPrevious"
      | "testRevert_BidMustMeetReservePrice"
      | "testRevert_CannotBidOnAuctionNotActive"
      | "testRevert_CannotBidOnAuctionNotStarted"
      | "testRevert_CannotBidOnExpiredAuction"
      | "testRevert_CannotCancelActiveAuction"
      | "testRevert_CannotUpdateActiveAuction"
      | "testRevert_CannotUpdateAuctionDoesNotExist"
      | "testRevert_FindersFeeBPSCannotExceed10000"
      | "testRevert_InvalidTransferBeforeFirstBid"
      | "testRevert_MustApproveModule"
      | "testRevert_MustBeTokenOwnerOrOperator"
      | "testRevert_MustSpecifySellerFundsRecipient"
      | "testRevert_OnlySellerOrOwnerCanCancel"
      | "testRevert_SellerMustApproveERC721TransferHelper"
      | "testRevert_UpdateMustBeSeller"
      | "test_CancelAuction"
      | "test_CreateAuction"
      | "test_CreateAuctionAndCancelPrevious"
      | "test_CreateFirstBid"
      | "test_CreateFutureAuction"
      | "test_ExtendAuction"
      | "test_RefundPreviousBidder"
      | "test_SetReservePrice"
      | "test_SettleAuction"
      | "test_StoreTimeOfFirstBid"
      | "test_TransferNFTIntoEscrow"
  ): FunctionFragment;

  encodeFunctionData(functionFragment: "IS_TEST", values?: undefined): string;
  encodeFunctionData(functionFragment: "failed", values?: undefined): string;
  encodeFunctionData(functionFragment: "setUp", values?: undefined): string;
  encodeFunctionData(
    functionFragment: "testRevert_AuctionNotOver",
    values?: undefined
  ): string;
  encodeFunctionData(
    functionFragment: "testRevert_AuctionNotStarted",
    values?: undefined
  ): string;
  encodeFunctionData(
    functionFragment: "testRevert_BidMustBe10PercentGreaterThanPrevious",
    values?: undefined
  ): string;
  encodeFunctionData(
    functionFragment: "testRevert_BidMustMeetReservePrice",
    values?: undefined
  ): string;
  encodeFunctionData(
    functionFragment: "testRevert_CannotBidOnAuctionNotActive",
    values?: undefined
  ): string;
  encodeFunctionData(
    functionFragment: "testRevert_CannotBidOnAuctionNotStarted",
    values?: undefined
  ): string;
  encodeFunctionData(
    functionFragment: "testRevert_CannotBidOnExpiredAuction",
    values?: undefined
  ): string;
  encodeFunctionData(
    functionFragment: "testRevert_CannotCancelActiveAuction",
    values?: undefined
  ): string;
  encodeFunctionData(
    functionFragment: "testRevert_CannotUpdateActiveAuction",
    values?: undefined
  ): string;
  encodeFunctionData(
    functionFragment: "testRevert_CannotUpdateAuctionDoesNotExist",
    values?: undefined
  ): string;
  encodeFunctionData(
    functionFragment: "testRevert_FindersFeeBPSCannotExceed10000",
    values?: undefined
  ): string;
  encodeFunctionData(
    functionFragment: "testRevert_InvalidTransferBeforeFirstBid",
    values?: undefined
  ): string;
  encodeFunctionData(
    functionFragment: "testRevert_MustApproveModule",
    values?: undefined
  ): string;
  encodeFunctionData(
    functionFragment: "testRevert_MustBeTokenOwnerOrOperator",
    values?: undefined
  ): string;
  encodeFunctionData(
    functionFragment: "testRevert_MustSpecifySellerFundsRecipient",
    values?: undefined
  ): string;
  encodeFunctionData(
    functionFragment: "testRevert_OnlySellerOrOwnerCanCancel",
    values?: undefined
  ): string;
  encodeFunctionData(
    functionFragment: "testRevert_SellerMustApproveERC721TransferHelper",
    values?: undefined
  ): string;
  encodeFunctionData(
    functionFragment: "testRevert_UpdateMustBeSeller",
    values?: undefined
  ): string;
  encodeFunctionData(
    functionFragment: "test_CancelAuction",
    values?: undefined
  ): string;
  encodeFunctionData(
    functionFragment: "test_CreateAuction",
    values?: undefined
  ): string;
  encodeFunctionData(
    functionFragment: "test_CreateAuctionAndCancelPrevious",
    values?: undefined
  ): string;
  encodeFunctionData(
    functionFragment: "test_CreateFirstBid",
    values?: undefined
  ): string;
  encodeFunctionData(
    functionFragment: "test_CreateFutureAuction",
    values?: undefined
  ): string;
  encodeFunctionData(
    functionFragment: "test_ExtendAuction",
    values?: undefined
  ): string;
  encodeFunctionData(
    functionFragment: "test_RefundPreviousBidder",
    values?: undefined
  ): string;
  encodeFunctionData(
    functionFragment: "test_SetReservePrice",
    values?: undefined
  ): string;
  encodeFunctionData(
    functionFragment: "test_SettleAuction",
    values?: undefined
  ): string;
  encodeFunctionData(
    functionFragment: "test_StoreTimeOfFirstBid",
    values?: undefined
  ): string;
  encodeFunctionData(
    functionFragment: "test_TransferNFTIntoEscrow",
    values?: undefined
  ): string;

  decodeFunctionResult(functionFragment: "IS_TEST", data: BytesLike): Result;
  decodeFunctionResult(functionFragment: "failed", data: BytesLike): Result;
  decodeFunctionResult(functionFragment: "setUp", data: BytesLike): Result;
  decodeFunctionResult(
    functionFragment: "testRevert_AuctionNotOver",
    data: BytesLike
  ): Result;
  decodeFunctionResult(
    functionFragment: "testRevert_AuctionNotStarted",
    data: BytesLike
  ): Result;
  decodeFunctionResult(
    functionFragment: "testRevert_BidMustBe10PercentGreaterThanPrevious",
    data: BytesLike
  ): Result;
  decodeFunctionResult(
    functionFragment: "testRevert_BidMustMeetReservePrice",
    data: BytesLike
  ): Result;
  decodeFunctionResult(
    functionFragment: "testRevert_CannotBidOnAuctionNotActive",
    data: BytesLike
  ): Result;
  decodeFunctionResult(
    functionFragment: "testRevert_CannotBidOnAuctionNotStarted",
    data: BytesLike
  ): Result;
  decodeFunctionResult(
    functionFragment: "testRevert_CannotBidOnExpiredAuction",
    data: BytesLike
  ): Result;
  decodeFunctionResult(
    functionFragment: "testRevert_CannotCancelActiveAuction",
    data: BytesLike
  ): Result;
  decodeFunctionResult(
    functionFragment: "testRevert_CannotUpdateActiveAuction",
    data: BytesLike
  ): Result;
  decodeFunctionResult(
    functionFragment: "testRevert_CannotUpdateAuctionDoesNotExist",
    data: BytesLike
  ): Result;
  decodeFunctionResult(
    functionFragment: "testRevert_FindersFeeBPSCannotExceed10000",
    data: BytesLike
  ): Result;
  decodeFunctionResult(
    functionFragment: "testRevert_InvalidTransferBeforeFirstBid",
    data: BytesLike
  ): Result;
  decodeFunctionResult(
    functionFragment: "testRevert_MustApproveModule",
    data: BytesLike
  ): Result;
  decodeFunctionResult(
    functionFragment: "testRevert_MustBeTokenOwnerOrOperator",
    data: BytesLike
  ): Result;
  decodeFunctionResult(
    functionFragment: "testRevert_MustSpecifySellerFundsRecipient",
    data: BytesLike
  ): Result;
  decodeFunctionResult(
    functionFragment: "testRevert_OnlySellerOrOwnerCanCancel",
    data: BytesLike
  ): Result;
  decodeFunctionResult(
    functionFragment: "testRevert_SellerMustApproveERC721TransferHelper",
    data: BytesLike
  ): Result;
  decodeFunctionResult(
    functionFragment: "testRevert_UpdateMustBeSeller",
    data: BytesLike
  ): Result;
  decodeFunctionResult(
    functionFragment: "test_CancelAuction",
    data: BytesLike
  ): Result;
  decodeFunctionResult(
    functionFragment: "test_CreateAuction",
    data: BytesLike
  ): Result;
  decodeFunctionResult(
    functionFragment: "test_CreateAuctionAndCancelPrevious",
    data: BytesLike
  ): Result;
  decodeFunctionResult(
    functionFragment: "test_CreateFirstBid",
    data: BytesLike
  ): Result;
  decodeFunctionResult(
    functionFragment: "test_CreateFutureAuction",
    data: BytesLike
  ): Result;
  decodeFunctionResult(
    functionFragment: "test_ExtendAuction",
    data: BytesLike
  ): Result;
  decodeFunctionResult(
    functionFragment: "test_RefundPreviousBidder",
    data: BytesLike
  ): Result;
  decodeFunctionResult(
    functionFragment: "test_SetReservePrice",
    data: BytesLike
  ): Result;
  decodeFunctionResult(
    functionFragment: "test_SettleAuction",
    data: BytesLike
  ): Result;
  decodeFunctionResult(
    functionFragment: "test_StoreTimeOfFirstBid",
    data: BytesLike
  ): Result;
  decodeFunctionResult(
    functionFragment: "test_TransferNFTIntoEscrow",
    data: BytesLike
  ): Result;

  events: {
    "log(string)": EventFragment;
    "log_address(address)": EventFragment;
    "log_bytes(bytes)": EventFragment;
    "log_bytes32(bytes32)": EventFragment;
    "log_int(int256)": EventFragment;
    "log_named_address(string,address)": EventFragment;
    "log_named_bytes(string,bytes)": EventFragment;
    "log_named_bytes32(string,bytes32)": EventFragment;
    "log_named_decimal_int(string,int256,uint256)": EventFragment;
    "log_named_decimal_uint(string,uint256,uint256)": EventFragment;
    "log_named_int(string,int256)": EventFragment;
    "log_named_string(string,string)": EventFragment;
    "log_named_uint(string,uint256)": EventFragment;
    "log_string(string)": EventFragment;
    "log_uint(uint256)": EventFragment;
    "logs(bytes)": EventFragment;
  };

  getEvent(nameOrSignatureOrTopic: "log"): EventFragment;
  getEvent(nameOrSignatureOrTopic: "log_address"): EventFragment;
  getEvent(nameOrSignatureOrTopic: "log_bytes"): EventFragment;
  getEvent(nameOrSignatureOrTopic: "log_bytes32"): EventFragment;
  getEvent(nameOrSignatureOrTopic: "log_int"): EventFragment;
  getEvent(nameOrSignatureOrTopic: "log_named_address"): EventFragment;
  getEvent(nameOrSignatureOrTopic: "log_named_bytes"): EventFragment;
  getEvent(nameOrSignatureOrTopic: "log_named_bytes32"): EventFragment;
  getEvent(nameOrSignatureOrTopic: "log_named_decimal_int"): EventFragment;
  getEvent(nameOrSignatureOrTopic: "log_named_decimal_uint"): EventFragment;
  getEvent(nameOrSignatureOrTopic: "log_named_int"): EventFragment;
  getEvent(nameOrSignatureOrTopic: "log_named_string"): EventFragment;
  getEvent(nameOrSignatureOrTopic: "log_named_uint"): EventFragment;
  getEvent(nameOrSignatureOrTopic: "log_string"): EventFragment;
  getEvent(nameOrSignatureOrTopic: "log_uint"): EventFragment;
  getEvent(nameOrSignatureOrTopic: "logs"): EventFragment;
}

export interface logEventObject {
  arg0: string;
}
export type logEvent = TypedEvent<[string], logEventObject>;

export type logEventFilter = TypedEventFilter<logEvent>;

export interface log_addressEventObject {
  arg0: string;
}
export type log_addressEvent = TypedEvent<[string], log_addressEventObject>;

export type log_addressEventFilter = TypedEventFilter<log_addressEvent>;

export interface log_bytesEventObject {
  arg0: string;
}
export type log_bytesEvent = TypedEvent<[string], log_bytesEventObject>;

export type log_bytesEventFilter = TypedEventFilter<log_bytesEvent>;

export interface log_bytes32EventObject {
  arg0: string;
}
export type log_bytes32Event = TypedEvent<[string], log_bytes32EventObject>;

export type log_bytes32EventFilter = TypedEventFilter<log_bytes32Event>;

export interface log_intEventObject {
  arg0: BigNumber;
}
export type log_intEvent = TypedEvent<[BigNumber], log_intEventObject>;

export type log_intEventFilter = TypedEventFilter<log_intEvent>;

export interface log_named_addressEventObject {
  key: string;
  val: string;
}
export type log_named_addressEvent = TypedEvent<
  [string, string],
  log_named_addressEventObject
>;

export type log_named_addressEventFilter =
  TypedEventFilter<log_named_addressEvent>;

export interface log_named_bytesEventObject {
  key: string;
  val: string;
}
export type log_named_bytesEvent = TypedEvent<
  [string, string],
  log_named_bytesEventObject
>;

export type log_named_bytesEventFilter = TypedEventFilter<log_named_bytesEvent>;

export interface log_named_bytes32EventObject {
  key: string;
  val: string;
}
export type log_named_bytes32Event = TypedEvent<
  [string, string],
  log_named_bytes32EventObject
>;

export type log_named_bytes32EventFilter =
  TypedEventFilter<log_named_bytes32Event>;

export interface log_named_decimal_intEventObject {
  key: string;
  val: BigNumber;
  decimals: BigNumber;
}
export type log_named_decimal_intEvent = TypedEvent<
  [string, BigNumber, BigNumber],
  log_named_decimal_intEventObject
>;

export type log_named_decimal_intEventFilter =
  TypedEventFilter<log_named_decimal_intEvent>;

export interface log_named_decimal_uintEventObject {
  key: string;
  val: BigNumber;
  decimals: BigNumber;
}
export type log_named_decimal_uintEvent = TypedEvent<
  [string, BigNumber, BigNumber],
  log_named_decimal_uintEventObject
>;

export type log_named_decimal_uintEventFilter =
  TypedEventFilter<log_named_decimal_uintEvent>;

export interface log_named_intEventObject {
  key: string;
  val: BigNumber;
}
export type log_named_intEvent = TypedEvent<
  [string, BigNumber],
  log_named_intEventObject
>;

export type log_named_intEventFilter = TypedEventFilter<log_named_intEvent>;

export interface log_named_stringEventObject {
  key: string;
  val: string;
}
export type log_named_stringEvent = TypedEvent<
  [string, string],
  log_named_stringEventObject
>;

export type log_named_stringEventFilter =
  TypedEventFilter<log_named_stringEvent>;

export interface log_named_uintEventObject {
  key: string;
  val: BigNumber;
}
export type log_named_uintEvent = TypedEvent<
  [string, BigNumber],
  log_named_uintEventObject
>;

export type log_named_uintEventFilter = TypedEventFilter<log_named_uintEvent>;

export interface log_stringEventObject {
  arg0: string;
}
export type log_stringEvent = TypedEvent<[string], log_stringEventObject>;

export type log_stringEventFilter = TypedEventFilter<log_stringEvent>;

export interface log_uintEventObject {
  arg0: BigNumber;
}
export type log_uintEvent = TypedEvent<[BigNumber], log_uintEventObject>;

export type log_uintEventFilter = TypedEventFilter<log_uintEvent>;

export interface logsEventObject {
  arg0: string;
}
export type logsEvent = TypedEvent<[string], logsEventObject>;

export type logsEventFilter = TypedEventFilter<logsEvent>;

export interface ReserveAuctionFindersEthTest extends BaseContract {
  connect(signerOrProvider: Signer | Provider | string): this;
  attach(addressOrName: string): this;
  deployed(): Promise<this>;

  interface: ReserveAuctionFindersEthTestInterface;

  queryFilter<TEvent extends TypedEvent>(
    event: TypedEventFilter<TEvent>,
    fromBlockOrBlockhash?: string | number | undefined,
    toBlock?: string | number | undefined
  ): Promise<Array<TEvent>>;

  listeners<TEvent extends TypedEvent>(
    eventFilter?: TypedEventFilter<TEvent>
  ): Array<TypedListener<TEvent>>;
  listeners(eventName?: string): Array<Listener>;
  removeAllListeners<TEvent extends TypedEvent>(
    eventFilter: TypedEventFilter<TEvent>
  ): this;
  removeAllListeners(eventName?: string): this;
  off: OnEvent<this>;
  on: OnEvent<this>;
  once: OnEvent<this>;
  removeListener: OnEvent<this>;

  functions: {
    IS_TEST(overrides?: CallOverrides): Promise<[boolean]>;

    failed(overrides?: CallOverrides): Promise<[boolean]>;

    setUp(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<ContractTransaction>;

    testRevert_AuctionNotOver(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<ContractTransaction>;

    testRevert_AuctionNotStarted(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<ContractTransaction>;

    testRevert_BidMustBe10PercentGreaterThanPrevious(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<ContractTransaction>;

    testRevert_BidMustMeetReservePrice(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<ContractTransaction>;

    testRevert_CannotBidOnAuctionNotActive(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<ContractTransaction>;

    testRevert_CannotBidOnAuctionNotStarted(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<ContractTransaction>;

    testRevert_CannotBidOnExpiredAuction(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<ContractTransaction>;

    testRevert_CannotCancelActiveAuction(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<ContractTransaction>;

    testRevert_CannotUpdateActiveAuction(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<ContractTransaction>;

    testRevert_CannotUpdateAuctionDoesNotExist(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<ContractTransaction>;

    testRevert_FindersFeeBPSCannotExceed10000(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<ContractTransaction>;

    testRevert_InvalidTransferBeforeFirstBid(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<ContractTransaction>;

    testRevert_MustApproveModule(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<ContractTransaction>;

    testRevert_MustBeTokenOwnerOrOperator(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<ContractTransaction>;

    testRevert_MustSpecifySellerFundsRecipient(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<ContractTransaction>;

    testRevert_OnlySellerOrOwnerCanCancel(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<ContractTransaction>;

    testRevert_SellerMustApproveERC721TransferHelper(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<ContractTransaction>;

    testRevert_UpdateMustBeSeller(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<ContractTransaction>;

    test_CancelAuction(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<ContractTransaction>;

    test_CreateAuction(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<ContractTransaction>;

    test_CreateAuctionAndCancelPrevious(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<ContractTransaction>;

    test_CreateFirstBid(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<ContractTransaction>;

    test_CreateFutureAuction(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<ContractTransaction>;

    test_ExtendAuction(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<ContractTransaction>;

    test_RefundPreviousBidder(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<ContractTransaction>;

    test_SetReservePrice(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<ContractTransaction>;

    test_SettleAuction(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<ContractTransaction>;

    test_StoreTimeOfFirstBid(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<ContractTransaction>;

    test_TransferNFTIntoEscrow(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<ContractTransaction>;
  };

  IS_TEST(overrides?: CallOverrides): Promise<boolean>;

  failed(overrides?: CallOverrides): Promise<boolean>;

  setUp(
    overrides?: Overrides & { from?: PromiseOrValue<string> }
  ): Promise<ContractTransaction>;

  testRevert_AuctionNotOver(
    overrides?: Overrides & { from?: PromiseOrValue<string> }
  ): Promise<ContractTransaction>;

  testRevert_AuctionNotStarted(
    overrides?: Overrides & { from?: PromiseOrValue<string> }
  ): Promise<ContractTransaction>;

  testRevert_BidMustBe10PercentGreaterThanPrevious(
    overrides?: Overrides & { from?: PromiseOrValue<string> }
  ): Promise<ContractTransaction>;

  testRevert_BidMustMeetReservePrice(
    overrides?: Overrides & { from?: PromiseOrValue<string> }
  ): Promise<ContractTransaction>;

  testRevert_CannotBidOnAuctionNotActive(
    overrides?: Overrides & { from?: PromiseOrValue<string> }
  ): Promise<ContractTransaction>;

  testRevert_CannotBidOnAuctionNotStarted(
    overrides?: Overrides & { from?: PromiseOrValue<string> }
  ): Promise<ContractTransaction>;

  testRevert_CannotBidOnExpiredAuction(
    overrides?: Overrides & { from?: PromiseOrValue<string> }
  ): Promise<ContractTransaction>;

  testRevert_CannotCancelActiveAuction(
    overrides?: Overrides & { from?: PromiseOrValue<string> }
  ): Promise<ContractTransaction>;

  testRevert_CannotUpdateActiveAuction(
    overrides?: Overrides & { from?: PromiseOrValue<string> }
  ): Promise<ContractTransaction>;

  testRevert_CannotUpdateAuctionDoesNotExist(
    overrides?: Overrides & { from?: PromiseOrValue<string> }
  ): Promise<ContractTransaction>;

  testRevert_FindersFeeBPSCannotExceed10000(
    overrides?: Overrides & { from?: PromiseOrValue<string> }
  ): Promise<ContractTransaction>;

  testRevert_InvalidTransferBeforeFirstBid(
    overrides?: Overrides & { from?: PromiseOrValue<string> }
  ): Promise<ContractTransaction>;

  testRevert_MustApproveModule(
    overrides?: Overrides & { from?: PromiseOrValue<string> }
  ): Promise<ContractTransaction>;

  testRevert_MustBeTokenOwnerOrOperator(
    overrides?: Overrides & { from?: PromiseOrValue<string> }
  ): Promise<ContractTransaction>;

  testRevert_MustSpecifySellerFundsRecipient(
    overrides?: Overrides & { from?: PromiseOrValue<string> }
  ): Promise<ContractTransaction>;

  testRevert_OnlySellerOrOwnerCanCancel(
    overrides?: Overrides & { from?: PromiseOrValue<string> }
  ): Promise<ContractTransaction>;

  testRevert_SellerMustApproveERC721TransferHelper(
    overrides?: Overrides & { from?: PromiseOrValue<string> }
  ): Promise<ContractTransaction>;

  testRevert_UpdateMustBeSeller(
    overrides?: Overrides & { from?: PromiseOrValue<string> }
  ): Promise<ContractTransaction>;

  test_CancelAuction(
    overrides?: Overrides & { from?: PromiseOrValue<string> }
  ): Promise<ContractTransaction>;

  test_CreateAuction(
    overrides?: Overrides & { from?: PromiseOrValue<string> }
  ): Promise<ContractTransaction>;

  test_CreateAuctionAndCancelPrevious(
    overrides?: Overrides & { from?: PromiseOrValue<string> }
  ): Promise<ContractTransaction>;

  test_CreateFirstBid(
    overrides?: Overrides & { from?: PromiseOrValue<string> }
  ): Promise<ContractTransaction>;

  test_CreateFutureAuction(
    overrides?: Overrides & { from?: PromiseOrValue<string> }
  ): Promise<ContractTransaction>;

  test_ExtendAuction(
    overrides?: Overrides & { from?: PromiseOrValue<string> }
  ): Promise<ContractTransaction>;

  test_RefundPreviousBidder(
    overrides?: Overrides & { from?: PromiseOrValue<string> }
  ): Promise<ContractTransaction>;

  test_SetReservePrice(
    overrides?: Overrides & { from?: PromiseOrValue<string> }
  ): Promise<ContractTransaction>;

  test_SettleAuction(
    overrides?: Overrides & { from?: PromiseOrValue<string> }
  ): Promise<ContractTransaction>;

  test_StoreTimeOfFirstBid(
    overrides?: Overrides & { from?: PromiseOrValue<string> }
  ): Promise<ContractTransaction>;

  test_TransferNFTIntoEscrow(
    overrides?: Overrides & { from?: PromiseOrValue<string> }
  ): Promise<ContractTransaction>;

  callStatic: {
    IS_TEST(overrides?: CallOverrides): Promise<boolean>;

    failed(overrides?: CallOverrides): Promise<boolean>;

    setUp(overrides?: CallOverrides): Promise<void>;

    testRevert_AuctionNotOver(overrides?: CallOverrides): Promise<void>;

    testRevert_AuctionNotStarted(overrides?: CallOverrides): Promise<void>;

    testRevert_BidMustBe10PercentGreaterThanPrevious(
      overrides?: CallOverrides
    ): Promise<void>;

    testRevert_BidMustMeetReservePrice(
      overrides?: CallOverrides
    ): Promise<void>;

    testRevert_CannotBidOnAuctionNotActive(
      overrides?: CallOverrides
    ): Promise<void>;

    testRevert_CannotBidOnAuctionNotStarted(
      overrides?: CallOverrides
    ): Promise<void>;

    testRevert_CannotBidOnExpiredAuction(
      overrides?: CallOverrides
    ): Promise<void>;

    testRevert_CannotCancelActiveAuction(
      overrides?: CallOverrides
    ): Promise<void>;

    testRevert_CannotUpdateActiveAuction(
      overrides?: CallOverrides
    ): Promise<void>;

    testRevert_CannotUpdateAuctionDoesNotExist(
      overrides?: CallOverrides
    ): Promise<void>;

    testRevert_FindersFeeBPSCannotExceed10000(
      overrides?: CallOverrides
    ): Promise<void>;

    testRevert_InvalidTransferBeforeFirstBid(
      overrides?: CallOverrides
    ): Promise<void>;

    testRevert_MustApproveModule(overrides?: CallOverrides): Promise<void>;

    testRevert_MustBeTokenOwnerOrOperator(
      overrides?: CallOverrides
    ): Promise<void>;

    testRevert_MustSpecifySellerFundsRecipient(
      overrides?: CallOverrides
    ): Promise<void>;

    testRevert_OnlySellerOrOwnerCanCancel(
      overrides?: CallOverrides
    ): Promise<void>;

    testRevert_SellerMustApproveERC721TransferHelper(
      overrides?: CallOverrides
    ): Promise<void>;

    testRevert_UpdateMustBeSeller(overrides?: CallOverrides): Promise<void>;

    test_CancelAuction(overrides?: CallOverrides): Promise<void>;

    test_CreateAuction(overrides?: CallOverrides): Promise<void>;

    test_CreateAuctionAndCancelPrevious(
      overrides?: CallOverrides
    ): Promise<void>;

    test_CreateFirstBid(overrides?: CallOverrides): Promise<void>;

    test_CreateFutureAuction(overrides?: CallOverrides): Promise<void>;

    test_ExtendAuction(overrides?: CallOverrides): Promise<void>;

    test_RefundPreviousBidder(overrides?: CallOverrides): Promise<void>;

    test_SetReservePrice(overrides?: CallOverrides): Promise<void>;

    test_SettleAuction(overrides?: CallOverrides): Promise<void>;

    test_StoreTimeOfFirstBid(overrides?: CallOverrides): Promise<void>;

    test_TransferNFTIntoEscrow(overrides?: CallOverrides): Promise<void>;
  };

  filters: {
    "log(string)"(arg0?: null): logEventFilter;
    log(arg0?: null): logEventFilter;

    "log_address(address)"(arg0?: null): log_addressEventFilter;
    log_address(arg0?: null): log_addressEventFilter;

    "log_bytes(bytes)"(arg0?: null): log_bytesEventFilter;
    log_bytes(arg0?: null): log_bytesEventFilter;

    "log_bytes32(bytes32)"(arg0?: null): log_bytes32EventFilter;
    log_bytes32(arg0?: null): log_bytes32EventFilter;

    "log_int(int256)"(arg0?: null): log_intEventFilter;
    log_int(arg0?: null): log_intEventFilter;

    "log_named_address(string,address)"(
      key?: null,
      val?: null
    ): log_named_addressEventFilter;
    log_named_address(key?: null, val?: null): log_named_addressEventFilter;

    "log_named_bytes(string,bytes)"(
      key?: null,
      val?: null
    ): log_named_bytesEventFilter;
    log_named_bytes(key?: null, val?: null): log_named_bytesEventFilter;

    "log_named_bytes32(string,bytes32)"(
      key?: null,
      val?: null
    ): log_named_bytes32EventFilter;
    log_named_bytes32(key?: null, val?: null): log_named_bytes32EventFilter;

    "log_named_decimal_int(string,int256,uint256)"(
      key?: null,
      val?: null,
      decimals?: null
    ): log_named_decimal_intEventFilter;
    log_named_decimal_int(
      key?: null,
      val?: null,
      decimals?: null
    ): log_named_decimal_intEventFilter;

    "log_named_decimal_uint(string,uint256,uint256)"(
      key?: null,
      val?: null,
      decimals?: null
    ): log_named_decimal_uintEventFilter;
    log_named_decimal_uint(
      key?: null,
      val?: null,
      decimals?: null
    ): log_named_decimal_uintEventFilter;

    "log_named_int(string,int256)"(
      key?: null,
      val?: null
    ): log_named_intEventFilter;
    log_named_int(key?: null, val?: null): log_named_intEventFilter;

    "log_named_string(string,string)"(
      key?: null,
      val?: null
    ): log_named_stringEventFilter;
    log_named_string(key?: null, val?: null): log_named_stringEventFilter;

    "log_named_uint(string,uint256)"(
      key?: null,
      val?: null
    ): log_named_uintEventFilter;
    log_named_uint(key?: null, val?: null): log_named_uintEventFilter;

    "log_string(string)"(arg0?: null): log_stringEventFilter;
    log_string(arg0?: null): log_stringEventFilter;

    "log_uint(uint256)"(arg0?: null): log_uintEventFilter;
    log_uint(arg0?: null): log_uintEventFilter;

    "logs(bytes)"(arg0?: null): logsEventFilter;
    logs(arg0?: null): logsEventFilter;
  };

  estimateGas: {
    IS_TEST(overrides?: CallOverrides): Promise<BigNumber>;

    failed(overrides?: CallOverrides): Promise<BigNumber>;

    setUp(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<BigNumber>;

    testRevert_AuctionNotOver(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<BigNumber>;

    testRevert_AuctionNotStarted(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<BigNumber>;

    testRevert_BidMustBe10PercentGreaterThanPrevious(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<BigNumber>;

    testRevert_BidMustMeetReservePrice(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<BigNumber>;

    testRevert_CannotBidOnAuctionNotActive(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<BigNumber>;

    testRevert_CannotBidOnAuctionNotStarted(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<BigNumber>;

    testRevert_CannotBidOnExpiredAuction(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<BigNumber>;

    testRevert_CannotCancelActiveAuction(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<BigNumber>;

    testRevert_CannotUpdateActiveAuction(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<BigNumber>;

    testRevert_CannotUpdateAuctionDoesNotExist(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<BigNumber>;

    testRevert_FindersFeeBPSCannotExceed10000(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<BigNumber>;

    testRevert_InvalidTransferBeforeFirstBid(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<BigNumber>;

    testRevert_MustApproveModule(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<BigNumber>;

    testRevert_MustBeTokenOwnerOrOperator(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<BigNumber>;

    testRevert_MustSpecifySellerFundsRecipient(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<BigNumber>;

    testRevert_OnlySellerOrOwnerCanCancel(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<BigNumber>;

    testRevert_SellerMustApproveERC721TransferHelper(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<BigNumber>;

    testRevert_UpdateMustBeSeller(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<BigNumber>;

    test_CancelAuction(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<BigNumber>;

    test_CreateAuction(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<BigNumber>;

    test_CreateAuctionAndCancelPrevious(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<BigNumber>;

    test_CreateFirstBid(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<BigNumber>;

    test_CreateFutureAuction(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<BigNumber>;

    test_ExtendAuction(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<BigNumber>;

    test_RefundPreviousBidder(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<BigNumber>;

    test_SetReservePrice(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<BigNumber>;

    test_SettleAuction(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<BigNumber>;

    test_StoreTimeOfFirstBid(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<BigNumber>;

    test_TransferNFTIntoEscrow(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<BigNumber>;
  };

  populateTransaction: {
    IS_TEST(overrides?: CallOverrides): Promise<PopulatedTransaction>;

    failed(overrides?: CallOverrides): Promise<PopulatedTransaction>;

    setUp(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<PopulatedTransaction>;

    testRevert_AuctionNotOver(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<PopulatedTransaction>;

    testRevert_AuctionNotStarted(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<PopulatedTransaction>;

    testRevert_BidMustBe10PercentGreaterThanPrevious(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<PopulatedTransaction>;

    testRevert_BidMustMeetReservePrice(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<PopulatedTransaction>;

    testRevert_CannotBidOnAuctionNotActive(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<PopulatedTransaction>;

    testRevert_CannotBidOnAuctionNotStarted(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<PopulatedTransaction>;

    testRevert_CannotBidOnExpiredAuction(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<PopulatedTransaction>;

    testRevert_CannotCancelActiveAuction(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<PopulatedTransaction>;

    testRevert_CannotUpdateActiveAuction(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<PopulatedTransaction>;

    testRevert_CannotUpdateAuctionDoesNotExist(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<PopulatedTransaction>;

    testRevert_FindersFeeBPSCannotExceed10000(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<PopulatedTransaction>;

    testRevert_InvalidTransferBeforeFirstBid(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<PopulatedTransaction>;

    testRevert_MustApproveModule(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<PopulatedTransaction>;

    testRevert_MustBeTokenOwnerOrOperator(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<PopulatedTransaction>;

    testRevert_MustSpecifySellerFundsRecipient(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<PopulatedTransaction>;

    testRevert_OnlySellerOrOwnerCanCancel(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<PopulatedTransaction>;

    testRevert_SellerMustApproveERC721TransferHelper(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<PopulatedTransaction>;

    testRevert_UpdateMustBeSeller(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<PopulatedTransaction>;

    test_CancelAuction(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<PopulatedTransaction>;

    test_CreateAuction(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<PopulatedTransaction>;

    test_CreateAuctionAndCancelPrevious(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<PopulatedTransaction>;

    test_CreateFirstBid(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<PopulatedTransaction>;

    test_CreateFutureAuction(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<PopulatedTransaction>;

    test_ExtendAuction(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<PopulatedTransaction>;

    test_RefundPreviousBidder(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<PopulatedTransaction>;

    test_SetReservePrice(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<PopulatedTransaction>;

    test_SettleAuction(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<PopulatedTransaction>;

    test_StoreTimeOfFirstBid(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<PopulatedTransaction>;

    test_TransferNFTIntoEscrow(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<PopulatedTransaction>;
  };
}
