// @zoralabs/nft-hooks/dist/backends/zdk/utils/getStandardMarketData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getStandardMarketData = void 0;
const addresses_1 = require("../../../constants/addresses");
const dateToISO_1 = require("./dateToISO");
function getStandardMarketData({ market, amount, }) {
    var _a, _b;
    return {
        createdAt: {
            timestamp: (0, dateToISO_1.dateToISO)(market.transactionInfo.blockTimestamp),
            blockNumber: market.transactionInfo.blockNumber || undefined,
            transactionHash: market.transactionInfo.transactionHash || undefined,
        },
        amount: {
            usd: amount.usdcPrice
                ? {
                    raw: amount.usdcPrice.raw,
                    value: (_a = amount.usdcPrice) === null || _a === void 0 ? void 0 : _a.decimal,
                    currency: {
                        value: amount.nativePrice.decimal,
                        raw: amount.nativePrice.raw,
                    },
                }
                : undefined,
            eth: amount.nativePrice
                ? {
                    raw: (_b = amount.nativePrice) === null || _b === void 0 ? void 0 : _b.raw,
                    value: amount.nativePrice.decimal,
                    currency: {
                        address: addresses_1.ZERO_ADDRESS,
                        decimals: 18,
                        name: 'ETH',
                    },
                }
                : undefined,
            symbol: amount.nativePrice.currency.name,
            decimals: amount.nativePrice.currency.decimals,
            address: amount.nativePrice.currency.address,
            amount: {
                raw: amount.nativePrice.raw,
                decimal: amount.nativePrice.decimal,
                currency: amount.nativePrice.currency,
                value: amount.nativePrice.decimal,
                amount: {
                    raw: amount.nativePrice.raw,
                    value: amount.nativePrice.decimal,
                },
            },
        },
        raw: market,
    };
}
exports.getStandardMarketData = getStandardMarketData;
