// @zoralabs/nft-hooks/dist/backends/zora-indexer-v1/zora-indexer.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TOKENS_WITHOUT_AUCTIONS = exports.ACTIVE_AUCTIONS_QUERY = exports.BY_IDS = exports.BY_OWNER = void 0;
const graphql_request_1 = require("graphql-request");
const MEDIA_FRAGMENTS = (0, graphql_request_1.gql) `
  fragment V3AskPart on V3Ask {
    askPrice
    askCurrency
    status
    seller
    findersFeeBps
    finder
    events {
      eventType
      address
      transactionHash
      blockNumber
      blockTimestamp
    }
  }

  fragment IndexerTokenPart on Token {
    id
    tokenId
    owner
    address
    tokenContract {
      name
      symbol
      address
      supportsMetadata
    }
    tokenURI
    minter
    metadata {
      json
    }
    mintTransferEvent {
      transactionHash
      blockTimestamp
      blockNumber
    }
    media {
      contentURI
      contentHash
      metadataHash
      metadataURI
      ownerBidShare
      creatorBidShare
    }
    v3Ask {
      ...V3AskPart
    }
  }
  fragment AuctionBidEventPart on AuctionBidEvent {
    id
    value
    sender
    transactionHash
    blockNumber
    blockTimestamp
  }
  fragment IndexerAuctionPart on Auction {
    winner
    lastBidAmount
    duration
    tokenId
    auctionId
    approved
    tokenContract
    reservePrice
    firstBidTime
    expiresAt
    tokenOwner
    curator
    curatorFee
    curatorFeePercentage
    createdEvent {
      blockNumber
      blockTimestamp
      transactionHash
    }
    currency {
      name
      symbol
      decimals
      address
    }
    canceledEvent {
      transactionHash
      blockNumber
      blockTimestamp
    }
    endedEvent {
      transactionHash
      blockNumber
      blockTimestamp
    }
    bidEvents {
      ...AuctionBidEventPart
    }
  }
  fragment IndexerAuctionWithToken on Auction {
    ...IndexerAuctionPart
    token {
      ...IndexerTokenPart
    }
  }
`;
const TOKEN_TRANSFER_EVENTS = (0, graphql_request_1.gql) `
  fragment TokenTransferEventInfo on TokenTransferEvent {
    from
    to
    tokenId
    address
    blockTimestamp
    blockNumber
    transactionHash
  }
`;
const BASE_FRAGMENTS = (0, graphql_request_1.gql) `
  ${MEDIA_FRAGMENTS}
  ${TOKEN_TRANSFER_EVENTS}

  fragment IndexerTokenWithAuction on Token {
    ...IndexerTokenPart
    auctions {
      ...IndexerAuctionPart
    }
  }
`;
const DETAIL_FRAGMENTS = (0, graphql_request_1.gql) `
  fragment V3EventPart on Event {
    eventType
    address
    blockNumber
    blockTimestamp
    transactionHash
    details
  }

  fragment IndexerTokenWithAuctionDetail on Token {
    transferEvents {
      ...TokenTransferEventInfo
    }
    # Ask events
    v3Events {
      ...V3EventPart
    }
  }
`;
// Get list of nfts owned by user from contracts
exports.BY_OWNER = (0, graphql_request_1.gql) `
  ${BASE_FRAGMENTS}
  query byOwner(
    $addressQueryPart: String_comparison_exp!
    $owner: String
    $offset: Int
    $limit: Int
  ) @cached {
    Token(
      limit: $limit
      offset: $offset
      where: {
        address: $addressQueryPart
        _or: [
          { owner: { _eq: $owner } }
          {
            auctions: {
              _and: [
                { _not: { endedEvent: {} } }
                { _not: { canceledEvent: {} } }
                { tokenOwner: { _eq: $owner } }
              ]
            }
          }
        ]
      }
    ) {
      ...IndexerTokenWithAuction
    }
  }
`;
exports.BY_IDS = (0, graphql_request_1.gql) `
  ${BASE_FRAGMENTS}
  ${DETAIL_FRAGMENTS}

  query byIds($ids: [String!]) @cached {
    Token(where: { id: { _in: $ids } }) {
      ...IndexerTokenWithAuction
      ...IndexerTokenWithAuctionDetail
    }
  }
`;
exports.ACTIVE_AUCTIONS_QUERY = (0, graphql_request_1.gql) `
  ${BASE_FRAGMENTS}
  query activeAuctionsQuery(
    $andQuery: [Token_bool_exp!]
    $orderBy: [Token_order_by!]
    $limit: Int
    $offset: Int
  ) @cached {
    Token(
      where: { _and: $andQuery }
      order_by: $orderBy
      limit: $limit
      offset: $offset
    ) {
      ...IndexerTokenWithAuction
    }
  }
`;
exports.TOKENS_WITHOUT_AUCTIONS = (0, graphql_request_1.gql) `
  ${MEDIA_FRAGMENTS}
  query tokensWithoutAuctions($addresses: [String!], $limit: Int, $offset: Int) @cached {
    Token(
      where: {
        address: { _in: $addresses }
        _not: { auctions: {} }
        owner: { _neq: "0x0000000000000000000000000000000000000000" }
      }
      limit: $limit
      offset: $offset
    ) {
      ...IndexerTokenPart
    }
  }
`;
