// @zoralabs/v3/dist/typechain/Offers.t.sol/OffersV1Test.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
import type {
  BaseContract,
  BigNumber,
  BytesLike,
  CallOverrides,
  ContractTransaction,
  Overrides,
  PopulatedTransaction,
  Signer,
  utils,
} from "ethers";
import type {
  FunctionFragment,
  Result,
  EventFragment,
} from "@ethersproject/abi";
import type { Listener, Provider } from "@ethersproject/providers";
import type {
  TypedEventFilter,
  TypedEvent,
  TypedListener,
  OnEvent,
  PromiseOrValue,
} from "../common";

export interface OffersV1TestInterface extends utils.Interface {
  functions: {
    "IS_TEST()": FunctionFragment;
    "failed()": FunctionFragment;
    "setUp()": FunctionFragment;
    "testFail_CannotCreateOfferWithInvalidFindersFeeBps()": FunctionFragment;
    "testFail_CannotCreateOfferWithoutAttachingFunds()": FunctionFragment;
    "testGas_CreateOffer()": FunctionFragment;
    "testRevert_AcceptAmountMustMatchOffer()": FunctionFragment;
    "testRevert_AcceptCurrencyMustMatchOffer()": FunctionFragment;
    "testRevert_CannotCancelInactiveOffer()": FunctionFragment;
    "testRevert_CannotFillInactiveOffer()": FunctionFragment;
    "testRevert_CannotIncreaseOfferWithoutAttachingFunds()": FunctionFragment;
    "testRevert_CannotUpdateInactiveOffer()": FunctionFragment;
    "testRevert_CannotUpdateOfferWithPreviousAmount()": FunctionFragment;
    "testRevert_OnlySellerCanCancelOffer()": FunctionFragment;
    "testRevert_OnlySellerCanUpdateOffer()": FunctionFragment;
    "testRevert_OnlyTokenHolderCanFillOffer()": FunctionFragment;
    "test_CancelNFTOffer()": FunctionFragment;
    "test_CreateOffer()": FunctionFragment;
    "test_DecreaseETHOffer()": FunctionFragment;
    "test_DecreaseETHOfferWithERC20()": FunctionFragment;
    "test_FillNFTOffer()": FunctionFragment;
    "test_IncreaseETHOffer()": FunctionFragment;
    "test_IncreaseETHOfferWithERC20()": FunctionFragment;
  };

  getFunction(
    nameOrSignatureOrTopic:
      | "IS_TEST"
      | "failed"
      | "setUp"
      | "testFail_CannotCreateOfferWithInvalidFindersFeeBps"
      | "testFail_CannotCreateOfferWithoutAttachingFunds"
      | "testGas_CreateOffer"
      | "testRevert_AcceptAmountMustMatchOffer"
      | "testRevert_AcceptCurrencyMustMatchOffer"
      | "testRevert_CannotCancelInactiveOffer"
      | "testRevert_CannotFillInactiveOffer"
      | "testRevert_CannotIncreaseOfferWithoutAttachingFunds"
      | "testRevert_CannotUpdateInactiveOffer"
      | "testRevert_CannotUpdateOfferWithPreviousAmount"
      | "testRevert_OnlySellerCanCancelOffer"
      | "testRevert_OnlySellerCanUpdateOffer"
      | "testRevert_OnlyTokenHolderCanFillOffer"
      | "test_CancelNFTOffer"
      | "test_CreateOffer"
      | "test_DecreaseETHOffer"
      | "test_DecreaseETHOfferWithERC20"
      | "test_FillNFTOffer"
      | "test_IncreaseETHOffer"
      | "test_IncreaseETHOfferWithERC20"
  ): FunctionFragment;

  encodeFunctionData(functionFragment: "IS_TEST", values?: undefined): string;
  encodeFunctionData(functionFragment: "failed", values?: undefined): string;
  encodeFunctionData(functionFragment: "setUp", values?: undefined): string;
  encodeFunctionData(
    functionFragment: "testFail_CannotCreateOfferWithInvalidFindersFeeBps",
    values?: undefined
  ): string;
  encodeFunctionData(
    functionFragment: "testFail_CannotCreateOfferWithoutAttachingFunds",
    values?: undefined
  ): string;
  encodeFunctionData(
    functionFragment: "testGas_CreateOffer",
    values?: undefined
  ): string;
  encodeFunctionData(
    functionFragment: "testRevert_AcceptAmountMustMatchOffer",
    values?: undefined
  ): string;
  encodeFunctionData(
    functionFragment: "testRevert_AcceptCurrencyMustMatchOffer",
    values?: undefined
  ): string;
  encodeFunctionData(
    functionFragment: "testRevert_CannotCancelInactiveOffer",
    values?: undefined
  ): string;
  encodeFunctionData(
    functionFragment: "testRevert_CannotFillInactiveOffer",
    values?: undefined
  ): string;
  encodeFunctionData(
    functionFragment: "testRevert_CannotIncreaseOfferWithoutAttachingFunds",
    values?: undefined
  ): string;
  encodeFunctionData(
    functionFragment: "testRevert_CannotUpdateInactiveOffer",
    values?: undefined
  ): string;
  encodeFunctionData(
    functionFragment: "testRevert_CannotUpdateOfferWithPreviousAmount",
    values?: undefined
  ): string;
  encodeFunctionData(
    functionFragment: "testRevert_OnlySellerCanCancelOffer",
    values?: undefined
  ): string;
  encodeFunctionData(
    functionFragment: "testRevert_OnlySellerCanUpdateOffer",
    values?: undefined
  ): string;
  encodeFunctionData(
    functionFragment: "testRevert_OnlyTokenHolderCanFillOffer",
    values?: undefined
  ): string;
  encodeFunctionData(
    functionFragment: "test_CancelNFTOffer",
    values?: undefined
  ): string;
  encodeFunctionData(
    functionFragment: "test_CreateOffer",
    values?: undefined
  ): string;
  encodeFunctionData(
    functionFragment: "test_DecreaseETHOffer",
    values?: undefined
  ): string;
  encodeFunctionData(
    functionFragment: "test_DecreaseETHOfferWithERC20",
    values?: undefined
  ): string;
  encodeFunctionData(
    functionFragment: "test_FillNFTOffer",
    values?: undefined
  ): string;
  encodeFunctionData(
    functionFragment: "test_IncreaseETHOffer",
    values?: undefined
  ): string;
  encodeFunctionData(
    functionFragment: "test_IncreaseETHOfferWithERC20",
    values?: undefined
  ): string;

  decodeFunctionResult(functionFragment: "IS_TEST", data: BytesLike): Result;
  decodeFunctionResult(functionFragment: "failed", data: BytesLike): Result;
  decodeFunctionResult(functionFragment: "setUp", data: BytesLike): Result;
  decodeFunctionResult(
    functionFragment: "testFail_CannotCreateOfferWithInvalidFindersFeeBps",
    data: BytesLike
  ): Result;
  decodeFunctionResult(
    functionFragment: "testFail_CannotCreateOfferWithoutAttachingFunds",
    data: BytesLike
  ): Result;
  decodeFunctionResult(
    functionFragment: "testGas_CreateOffer",
    data: BytesLike
  ): Result;
  decodeFunctionResult(
    functionFragment: "testRevert_AcceptAmountMustMatchOffer",
    data: BytesLike
  ): Result;
  decodeFunctionResult(
    functionFragment: "testRevert_AcceptCurrencyMustMatchOffer",
    data: BytesLike
  ): Result;
  decodeFunctionResult(
    functionFragment: "testRevert_CannotCancelInactiveOffer",
    data: BytesLike
  ): Result;
  decodeFunctionResult(
    functionFragment: "testRevert_CannotFillInactiveOffer",
    data: BytesLike
  ): Result;
  decodeFunctionResult(
    functionFragment: "testRevert_CannotIncreaseOfferWithoutAttachingFunds",
    data: BytesLike
  ): Result;
  decodeFunctionResult(
    functionFragment: "testRevert_CannotUpdateInactiveOffer",
    data: BytesLike
  ): Result;
  decodeFunctionResult(
    functionFragment: "testRevert_CannotUpdateOfferWithPreviousAmount",
    data: BytesLike
  ): Result;
  decodeFunctionResult(
    functionFragment: "testRevert_OnlySellerCanCancelOffer",
    data: BytesLike
  ): Result;
  decodeFunctionResult(
    functionFragment: "testRevert_OnlySellerCanUpdateOffer",
    data: BytesLike
  ): Result;
  decodeFunctionResult(
    functionFragment: "testRevert_OnlyTokenHolderCanFillOffer",
    data: BytesLike
  ): Result;
  decodeFunctionResult(
    functionFragment: "test_CancelNFTOffer",
    data: BytesLike
  ): Result;
  decodeFunctionResult(
    functionFragment: "test_CreateOffer",
    data: BytesLike
  ): Result;
  decodeFunctionResult(
    functionFragment: "test_DecreaseETHOffer",
    data: BytesLike
  ): Result;
  decodeFunctionResult(
    functionFragment: "test_DecreaseETHOfferWithERC20",
    data: BytesLike
  ): Result;
  decodeFunctionResult(
    functionFragment: "test_FillNFTOffer",
    data: BytesLike
  ): Result;
  decodeFunctionResult(
    functionFragment: "test_IncreaseETHOffer",
    data: BytesLike
  ): Result;
  decodeFunctionResult(
    functionFragment: "test_IncreaseETHOfferWithERC20",
    data: BytesLike
  ): Result;

  events: {
    "log(string)": EventFragment;
    "log_address(address)": EventFragment;
    "log_bytes(bytes)": EventFragment;
    "log_bytes32(bytes32)": EventFragment;
    "log_int(int256)": EventFragment;
    "log_named_address(string,address)": EventFragment;
    "log_named_bytes(string,bytes)": EventFragment;
    "log_named_bytes32(string,bytes32)": EventFragment;
    "log_named_decimal_int(string,int256,uint256)": EventFragment;
    "log_named_decimal_uint(string,uint256,uint256)": EventFragment;
    "log_named_int(string,int256)": EventFragment;
    "log_named_string(string,string)": EventFragment;
    "log_named_uint(string,uint256)": EventFragment;
    "log_string(string)": EventFragment;
    "log_uint(uint256)": EventFragment;
    "logs(bytes)": EventFragment;
  };

  getEvent(nameOrSignatureOrTopic: "log"): EventFragment;
  getEvent(nameOrSignatureOrTopic: "log_address"): EventFragment;
  getEvent(nameOrSignatureOrTopic: "log_bytes"): EventFragment;
  getEvent(nameOrSignatureOrTopic: "log_bytes32"): EventFragment;
  getEvent(nameOrSignatureOrTopic: "log_int"): EventFragment;
  getEvent(nameOrSignatureOrTopic: "log_named_address"): EventFragment;
  getEvent(nameOrSignatureOrTopic: "log_named_bytes"): EventFragment;
  getEvent(nameOrSignatureOrTopic: "log_named_bytes32"): EventFragment;
  getEvent(nameOrSignatureOrTopic: "log_named_decimal_int"): EventFragment;
  getEvent(nameOrSignatureOrTopic: "log_named_decimal_uint"): EventFragment;
  getEvent(nameOrSignatureOrTopic: "log_named_int"): EventFragment;
  getEvent(nameOrSignatureOrTopic: "log_named_string"): EventFragment;
  getEvent(nameOrSignatureOrTopic: "log_named_uint"): EventFragment;
  getEvent(nameOrSignatureOrTopic: "log_string"): EventFragment;
  getEvent(nameOrSignatureOrTopic: "log_uint"): EventFragment;
  getEvent(nameOrSignatureOrTopic: "logs"): EventFragment;
}

export interface logEventObject {
  arg0: string;
}
export type logEvent = TypedEvent<[string], logEventObject>;

export type logEventFilter = TypedEventFilter<logEvent>;

export interface log_addressEventObject {
  arg0: string;
}
export type log_addressEvent = TypedEvent<[string], log_addressEventObject>;

export type log_addressEventFilter = TypedEventFilter<log_addressEvent>;

export interface log_bytesEventObject {
  arg0: string;
}
export type log_bytesEvent = TypedEvent<[string], log_bytesEventObject>;

export type log_bytesEventFilter = TypedEventFilter<log_bytesEvent>;

export interface log_bytes32EventObject {
  arg0: string;
}
export type log_bytes32Event = TypedEvent<[string], log_bytes32EventObject>;

export type log_bytes32EventFilter = TypedEventFilter<log_bytes32Event>;

export interface log_intEventObject {
  arg0: BigNumber;
}
export type log_intEvent = TypedEvent<[BigNumber], log_intEventObject>;

export type log_intEventFilter = TypedEventFilter<log_intEvent>;

export interface log_named_addressEventObject {
  key: string;
  val: string;
}
export type log_named_addressEvent = TypedEvent<
  [string, string],
  log_named_addressEventObject
>;

export type log_named_addressEventFilter =
  TypedEventFilter<log_named_addressEvent>;

export interface log_named_bytesEventObject {
  key: string;
  val: string;
}
export type log_named_bytesEvent = TypedEvent<
  [string, string],
  log_named_bytesEventObject
>;

export type log_named_bytesEventFilter = TypedEventFilter<log_named_bytesEvent>;

export interface log_named_bytes32EventObject {
  key: string;
  val: string;
}
export type log_named_bytes32Event = TypedEvent<
  [string, string],
  log_named_bytes32EventObject
>;

export type log_named_bytes32EventFilter =
  TypedEventFilter<log_named_bytes32Event>;

export interface log_named_decimal_intEventObject {
  key: string;
  val: BigNumber;
  decimals: BigNumber;
}
export type log_named_decimal_intEvent = TypedEvent<
  [string, BigNumber, BigNumber],
  log_named_decimal_intEventObject
>;

export type log_named_decimal_intEventFilter =
  TypedEventFilter<log_named_decimal_intEvent>;

export interface log_named_decimal_uintEventObject {
  key: string;
  val: BigNumber;
  decimals: BigNumber;
}
export type log_named_decimal_uintEvent = TypedEvent<
  [string, BigNumber, BigNumber],
  log_named_decimal_uintEventObject
>;

export type log_named_decimal_uintEventFilter =
  TypedEventFilter<log_named_decimal_uintEvent>;

export interface log_named_intEventObject {
  key: string;
  val: BigNumber;
}
export type log_named_intEvent = TypedEvent<
  [string, BigNumber],
  log_named_intEventObject
>;

export type log_named_intEventFilter = TypedEventFilter<log_named_intEvent>;

export interface log_named_stringEventObject {
  key: string;
  val: string;
}
export type log_named_stringEvent = TypedEvent<
  [string, string],
  log_named_stringEventObject
>;

export type log_named_stringEventFilter =
  TypedEventFilter<log_named_stringEvent>;

export interface log_named_uintEventObject {
  key: string;
  val: BigNumber;
}
export type log_named_uintEvent = TypedEvent<
  [string, BigNumber],
  log_named_uintEventObject
>;

export type log_named_uintEventFilter = TypedEventFilter<log_named_uintEvent>;

export interface log_stringEventObject {
  arg0: string;
}
export type log_stringEvent = TypedEvent<[string], log_stringEventObject>;

export type log_stringEventFilter = TypedEventFilter<log_stringEvent>;

export interface log_uintEventObject {
  arg0: BigNumber;
}
export type log_uintEvent = TypedEvent<[BigNumber], log_uintEventObject>;

export type log_uintEventFilter = TypedEventFilter<log_uintEvent>;

export interface logsEventObject {
  arg0: string;
}
export type logsEvent = TypedEvent<[string], logsEventObject>;

export type logsEventFilter = TypedEventFilter<logsEvent>;

export interface OffersV1Test extends BaseContract {
  connect(signerOrProvider: Signer | Provider | string): this;
  attach(addressOrName: string): this;
  deployed(): Promise<this>;

  interface: OffersV1TestInterface;

  queryFilter<TEvent extends TypedEvent>(
    event: TypedEventFilter<TEvent>,
    fromBlockOrBlockhash?: string | number | undefined,
    toBlock?: string | number | undefined
  ): Promise<Array<TEvent>>;

  listeners<TEvent extends TypedEvent>(
    eventFilter?: TypedEventFilter<TEvent>
  ): Array<TypedListener<TEvent>>;
  listeners(eventName?: string): Array<Listener>;
  removeAllListeners<TEvent extends TypedEvent>(
    eventFilter: TypedEventFilter<TEvent>
  ): this;
  removeAllListeners(eventName?: string): this;
  off: OnEvent<this>;
  on: OnEvent<this>;
  once: OnEvent<this>;
  removeListener: OnEvent<this>;

  functions: {
    IS_TEST(overrides?: CallOverrides): Promise<[boolean]>;

    failed(overrides?: CallOverrides): Promise<[boolean]>;

    setUp(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<ContractTransaction>;

    testFail_CannotCreateOfferWithInvalidFindersFeeBps(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<ContractTransaction>;

    testFail_CannotCreateOfferWithoutAttachingFunds(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<ContractTransaction>;

    testGas_CreateOffer(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<ContractTransaction>;

    testRevert_AcceptAmountMustMatchOffer(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<ContractTransaction>;

    testRevert_AcceptCurrencyMustMatchOffer(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<ContractTransaction>;

    testRevert_CannotCancelInactiveOffer(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<ContractTransaction>;

    testRevert_CannotFillInactiveOffer(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<ContractTransaction>;

    testRevert_CannotIncreaseOfferWithoutAttachingFunds(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<ContractTransaction>;

    testRevert_CannotUpdateInactiveOffer(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<ContractTransaction>;

    testRevert_CannotUpdateOfferWithPreviousAmount(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<ContractTransaction>;

    testRevert_OnlySellerCanCancelOffer(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<ContractTransaction>;

    testRevert_OnlySellerCanUpdateOffer(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<ContractTransaction>;

    testRevert_OnlyTokenHolderCanFillOffer(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<ContractTransaction>;

    test_CancelNFTOffer(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<ContractTransaction>;

    test_CreateOffer(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<ContractTransaction>;

    test_DecreaseETHOffer(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<ContractTransaction>;

    test_DecreaseETHOfferWithERC20(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<ContractTransaction>;

    test_FillNFTOffer(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<ContractTransaction>;

    test_IncreaseETHOffer(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<ContractTransaction>;

    test_IncreaseETHOfferWithERC20(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<ContractTransaction>;
  };

  IS_TEST(overrides?: CallOverrides): Promise<boolean>;

  failed(overrides?: CallOverrides): Promise<boolean>;

  setUp(
    overrides?: Overrides & { from?: PromiseOrValue<string> }
  ): Promise<ContractTransaction>;

  testFail_CannotCreateOfferWithInvalidFindersFeeBps(
    overrides?: Overrides & { from?: PromiseOrValue<string> }
  ): Promise<ContractTransaction>;

  testFail_CannotCreateOfferWithoutAttachingFunds(
    overrides?: Overrides & { from?: PromiseOrValue<string> }
  ): Promise<ContractTransaction>;

  testGas_CreateOffer(
    overrides?: Overrides & { from?: PromiseOrValue<string> }
  ): Promise<ContractTransaction>;

  testRevert_AcceptAmountMustMatchOffer(
    overrides?: Overrides & { from?: PromiseOrValue<string> }
  ): Promise<ContractTransaction>;

  testRevert_AcceptCurrencyMustMatchOffer(
    overrides?: Overrides & { from?: PromiseOrValue<string> }
  ): Promise<ContractTransaction>;

  testRevert_CannotCancelInactiveOffer(
    overrides?: Overrides & { from?: PromiseOrValue<string> }
  ): Promise<ContractTransaction>;

  testRevert_CannotFillInactiveOffer(
    overrides?: Overrides & { from?: PromiseOrValue<string> }
  ): Promise<ContractTransaction>;

  testRevert_CannotIncreaseOfferWithoutAttachingFunds(
    overrides?: Overrides & { from?: PromiseOrValue<string> }
  ): Promise<ContractTransaction>;

  testRevert_CannotUpdateInactiveOffer(
    overrides?: Overrides & { from?: PromiseOrValue<string> }
  ): Promise<ContractTransaction>;

  testRevert_CannotUpdateOfferWithPreviousAmount(
    overrides?: Overrides & { from?: PromiseOrValue<string> }
  ): Promise<ContractTransaction>;

  testRevert_OnlySellerCanCancelOffer(
    overrides?: Overrides & { from?: PromiseOrValue<string> }
  ): Promise<ContractTransaction>;

  testRevert_OnlySellerCanUpdateOffer(
    overrides?: Overrides & { from?: PromiseOrValue<string> }
  ): Promise<ContractTransaction>;

  testRevert_OnlyTokenHolderCanFillOffer(
    overrides?: Overrides & { from?: PromiseOrValue<string> }
  ): Promise<ContractTransaction>;

  test_CancelNFTOffer(
    overrides?: Overrides & { from?: PromiseOrValue<string> }
  ): Promise<ContractTransaction>;

  test_CreateOffer(
    overrides?: Overrides & { from?: PromiseOrValue<string> }
  ): Promise<ContractTransaction>;

  test_DecreaseETHOffer(
    overrides?: Overrides & { from?: PromiseOrValue<string> }
  ): Promise<ContractTransaction>;

  test_DecreaseETHOfferWithERC20(
    overrides?: Overrides & { from?: PromiseOrValue<string> }
  ): Promise<ContractTransaction>;

  test_FillNFTOffer(
    overrides?: Overrides & { from?: PromiseOrValue<string> }
  ): Promise<ContractTransaction>;

  test_IncreaseETHOffer(
    overrides?: Overrides & { from?: PromiseOrValue<string> }
  ): Promise<ContractTransaction>;

  test_IncreaseETHOfferWithERC20(
    overrides?: Overrides & { from?: PromiseOrValue<string> }
  ): Promise<ContractTransaction>;

  callStatic: {
    IS_TEST(overrides?: CallOverrides): Promise<boolean>;

    failed(overrides?: CallOverrides): Promise<boolean>;

    setUp(overrides?: CallOverrides): Promise<void>;

    testFail_CannotCreateOfferWithInvalidFindersFeeBps(
      overrides?: CallOverrides
    ): Promise<void>;

    testFail_CannotCreateOfferWithoutAttachingFunds(
      overrides?: CallOverrides
    ): Promise<void>;

    testGas_CreateOffer(overrides?: CallOverrides): Promise<void>;

    testRevert_AcceptAmountMustMatchOffer(
      overrides?: CallOverrides
    ): Promise<void>;

    testRevert_AcceptCurrencyMustMatchOffer(
      overrides?: CallOverrides
    ): Promise<void>;

    testRevert_CannotCancelInactiveOffer(
      overrides?: CallOverrides
    ): Promise<void>;

    testRevert_CannotFillInactiveOffer(
      overrides?: CallOverrides
    ): Promise<void>;

    testRevert_CannotIncreaseOfferWithoutAttachingFunds(
      overrides?: CallOverrides
    ): Promise<void>;

    testRevert_CannotUpdateInactiveOffer(
      overrides?: CallOverrides
    ): Promise<void>;

    testRevert_CannotUpdateOfferWithPreviousAmount(
      overrides?: CallOverrides
    ): Promise<void>;

    testRevert_OnlySellerCanCancelOffer(
      overrides?: CallOverrides
    ): Promise<void>;

    testRevert_OnlySellerCanUpdateOffer(
      overrides?: CallOverrides
    ): Promise<void>;

    testRevert_OnlyTokenHolderCanFillOffer(
      overrides?: CallOverrides
    ): Promise<void>;

    test_CancelNFTOffer(overrides?: CallOverrides): Promise<void>;

    test_CreateOffer(overrides?: CallOverrides): Promise<void>;

    test_DecreaseETHOffer(overrides?: CallOverrides): Promise<void>;

    test_DecreaseETHOfferWithERC20(overrides?: CallOverrides): Promise<void>;

    test_FillNFTOffer(overrides?: CallOverrides): Promise<void>;

    test_IncreaseETHOffer(overrides?: CallOverrides): Promise<void>;

    test_IncreaseETHOfferWithERC20(overrides?: CallOverrides): Promise<void>;
  };

  filters: {
    "log(string)"(arg0?: null): logEventFilter;
    log(arg0?: null): logEventFilter;

    "log_address(address)"(arg0?: null): log_addressEventFilter;
    log_address(arg0?: null): log_addressEventFilter;

    "log_bytes(bytes)"(arg0?: null): log_bytesEventFilter;
    log_bytes(arg0?: null): log_bytesEventFilter;

    "log_bytes32(bytes32)"(arg0?: null): log_bytes32EventFilter;
    log_bytes32(arg0?: null): log_bytes32EventFilter;

    "log_int(int256)"(arg0?: null): log_intEventFilter;
    log_int(arg0?: null): log_intEventFilter;

    "log_named_address(string,address)"(
      key?: null,
      val?: null
    ): log_named_addressEventFilter;
    log_named_address(key?: null, val?: null): log_named_addressEventFilter;

    "log_named_bytes(string,bytes)"(
      key?: null,
      val?: null
    ): log_named_bytesEventFilter;
    log_named_bytes(key?: null, val?: null): log_named_bytesEventFilter;

    "log_named_bytes32(string,bytes32)"(
      key?: null,
      val?: null
    ): log_named_bytes32EventFilter;
    log_named_bytes32(key?: null, val?: null): log_named_bytes32EventFilter;

    "log_named_decimal_int(string,int256,uint256)"(
      key?: null,
      val?: null,
      decimals?: null
    ): log_named_decimal_intEventFilter;
    log_named_decimal_int(
      key?: null,
      val?: null,
      decimals?: null
    ): log_named_decimal_intEventFilter;

    "log_named_decimal_uint(string,uint256,uint256)"(
      key?: null,
      val?: null,
      decimals?: null
    ): log_named_decimal_uintEventFilter;
    log_named_decimal_uint(
      key?: null,
      val?: null,
      decimals?: null
    ): log_named_decimal_uintEventFilter;

    "log_named_int(string,int256)"(
      key?: null,
      val?: null
    ): log_named_intEventFilter;
    log_named_int(key?: null, val?: null): log_named_intEventFilter;

    "log_named_string(string,string)"(
      key?: null,
      val?: null
    ): log_named_stringEventFilter;
    log_named_string(key?: null, val?: null): log_named_stringEventFilter;

    "log_named_uint(string,uint256)"(
      key?: null,
      val?: null
    ): log_named_uintEventFilter;
    log_named_uint(key?: null, val?: null): log_named_uintEventFilter;

    "log_string(string)"(arg0?: null): log_stringEventFilter;
    log_string(arg0?: null): log_stringEventFilter;

    "log_uint(uint256)"(arg0?: null): log_uintEventFilter;
    log_uint(arg0?: null): log_uintEventFilter;

    "logs(bytes)"(arg0?: null): logsEventFilter;
    logs(arg0?: null): logsEventFilter;
  };

  estimateGas: {
    IS_TEST(overrides?: CallOverrides): Promise<BigNumber>;

    failed(overrides?: CallOverrides): Promise<BigNumber>;

    setUp(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<BigNumber>;

    testFail_CannotCreateOfferWithInvalidFindersFeeBps(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<BigNumber>;

    testFail_CannotCreateOfferWithoutAttachingFunds(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<BigNumber>;

    testGas_CreateOffer(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<BigNumber>;

    testRevert_AcceptAmountMustMatchOffer(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<BigNumber>;

    testRevert_AcceptCurrencyMustMatchOffer(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<BigNumber>;

    testRevert_CannotCancelInactiveOffer(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<BigNumber>;

    testRevert_CannotFillInactiveOffer(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<BigNumber>;

    testRevert_CannotIncreaseOfferWithoutAttachingFunds(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<BigNumber>;

    testRevert_CannotUpdateInactiveOffer(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<BigNumber>;

    testRevert_CannotUpdateOfferWithPreviousAmount(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<BigNumber>;

    testRevert_OnlySellerCanCancelOffer(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<BigNumber>;

    testRevert_OnlySellerCanUpdateOffer(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<BigNumber>;

    testRevert_OnlyTokenHolderCanFillOffer(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<BigNumber>;

    test_CancelNFTOffer(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<BigNumber>;

    test_CreateOffer(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<BigNumber>;

    test_DecreaseETHOffer(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<BigNumber>;

    test_DecreaseETHOfferWithERC20(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<BigNumber>;

    test_FillNFTOffer(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<BigNumber>;

    test_IncreaseETHOffer(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<BigNumber>;

    test_IncreaseETHOfferWithERC20(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<BigNumber>;
  };

  populateTransaction: {
    IS_TEST(overrides?: CallOverrides): Promise<PopulatedTransaction>;

    failed(overrides?: CallOverrides): Promise<PopulatedTransaction>;

    setUp(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<PopulatedTransaction>;

    testFail_CannotCreateOfferWithInvalidFindersFeeBps(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<PopulatedTransaction>;

    testFail_CannotCreateOfferWithoutAttachingFunds(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<PopulatedTransaction>;

    testGas_CreateOffer(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<PopulatedTransaction>;

    testRevert_AcceptAmountMustMatchOffer(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<PopulatedTransaction>;

    testRevert_AcceptCurrencyMustMatchOffer(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<PopulatedTransaction>;

    testRevert_CannotCancelInactiveOffer(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<PopulatedTransaction>;

    testRevert_CannotFillInactiveOffer(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<PopulatedTransaction>;

    testRevert_CannotIncreaseOfferWithoutAttachingFunds(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<PopulatedTransaction>;

    testRevert_CannotUpdateInactiveOffer(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<PopulatedTransaction>;

    testRevert_CannotUpdateOfferWithPreviousAmount(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<PopulatedTransaction>;

    testRevert_OnlySellerCanCancelOffer(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<PopulatedTransaction>;

    testRevert_OnlySellerCanUpdateOffer(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<PopulatedTransaction>;

    testRevert_OnlyTokenHolderCanFillOffer(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<PopulatedTransaction>;

    test_CancelNFTOffer(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<PopulatedTransaction>;

    test_CreateOffer(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<PopulatedTransaction>;

    test_DecreaseETHOffer(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<PopulatedTransaction>;

    test_DecreaseETHOfferWithERC20(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<PopulatedTransaction>;

    test_FillNFTOffer(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<PopulatedTransaction>;

    test_IncreaseETHOffer(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<PopulatedTransaction>;

    test_IncreaseETHOfferWithERC20(
      overrides?: Overrides & { from?: PromiseOrValue<string> }
    ): Promise<PopulatedTransaction>;
  };
}
